### PR TITLE
feat: Special support for XBlocks

### PR DESCRIPTION
**Pull request by shadinaif**
feat: Special support for XBlocks

XBlocks require `text.po` and `textjs.po` instead of `django.po` and `djangojs.po`. This PR will add support to find XBlock translation files and rename them to `django` during translation extraction.

**Why is this needed?**

After moving along with standardizing `make extract_translations` in XBlocks; we found that it's safer to keep `text.po` and `textjs.po` as the translation domains for XBlocks. It's to preserve compatibility with in-repo translations since it'll be used for a while by the community

See this related conversation https://github.com/openedx/edx-cookiecutters/pull/382#discussion_r1300041928

- [x] Tested in a forked PR https://github.com/Zeit-Labs/openedx-translations/pull/67/files. The tested XBlock is a fork containing `text.po` as the domain for translation https://github.com/Zeit-Labs/xblock-drag-and-drop-v2/blob/9956abde00175e3d53dff53502b10fafbd6af0d3/Makefile#L12

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @shadinaif!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by shadinaif** [comment]
This is ready @OmarIthawi @brian-smith-tcril

**Review by brian-smith-tcril — APPROVED** [review]
This LGTM, I'll wait to see if @OmarIthawi has any feedback before merging.

**Review by OmarIthawi — APPROVED** [review]
@shadinaif I agree with the overall process. However, the implementation is becoming a bit cumbersome.

Nevertheless, I think we should refactor this file eventually. Therefore, I've added an issue for that:

 - https://github.com/openedx/openedx-translations/issues/606

File: .github/workflows/extract-translation-source-files.yml
Comment (OmarIthawi) [inline .github/workflows/extract-translation-source-files.yml:158]:
```
$ man test
NAME
       test - check file types and compare values
...
       -n STRING
              the length of STRING is nonzero
...
```


Therefore the nitpick: 
```suggestion
            if [ -n "$DJANGO_PATH" ]; then
```

File: .github/workflows/extract-translation-source-files.yml
Comment (OmarIthawi) [inline .github/workflows/extract-translation-source-files.yml:162]: Same here.

```suggestion
            if [ -n "$DJANGOJS_PATH" ]; then
```

Comment (shadinaif) [reply .github/workflows/extract-translation-source-files.yml]: Done

**Comment by openedx-webhooks** [comment]
@shadinaif 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-translations&entry.867055334=https://github.com/openedx/openedx-translations/pull/593&entry.1484655318=https://github.com/shadinaif&entry.752974735=2023-08-23+12:02&entry.1917517419=2023-08-31+09:17&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
